docs/core-functionality.md

```diff
@@ -0,0 +1,17 @@
+# Core functionality
```

Comment:
If there is a separate section for core functionality, then things like pipeline etc should also be described here right?